### ISSUE TITLE
Adding a new folder doesnt move cursor to second (inactive) window

### ISSUE THREAD

**Issue by DPC15038** [microsoft/vscode#199976]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.84.2 (Universal)
- OS Version: MacOS 14.0 (23A344)

Steps to Reproduce:

1. Have two windows open side by side with both File Explorers open
2. In the right window, expand a folder in the File Explorer.
3. In the left window, right click any folder and click either "Create File" or "Create Folder"
4. As you start to type, you'll notice the left window isnt active (even though you just clicked within it), so you're still typing in the right window.

This typically happens when re-creating a similar file organizational structure between two repos, and is unbelievably annoying.

**Comment by nachocab** [comment]
I've also noticed this for the past few weeks. It happens intermittently, so I'm not able to find a reliable way to reproduce it. 

What I've noticed is that I sometimes click the new file icon but the window doesn't come into focus:
![CleanShot 2023-12-05 at 16 56 07@2x](https://github.com/microsoft/vscode/assets/27538/683de1fe-a995-4758-b5ae-bef001e03bb3)


The new filename input doesn't have the blinking cursor:
![CleanShot 2023-12-05 at 16 56 35@2x](https://github.com/microsoft/vscode/assets/27538/858a3ccf-b7fa-4271-9973-a1985ea9aa56)

I start typing, but nothing happens because it's not on focus. When I click on the window, the filename input disappears. If I click the new file button again, this time it works because it's in focus:

![CleanShot 2023-12-05 at 16 55 40@2x](https://github.com/microsoft/vscode/assets/27538/2b4f1273-f8a4-4c9e-87e2-558bc9d498da)

> 1. Have two windows open side by side with both File Explorers open
I almost always have multiple VSCode windows open, so that might be a hint.
